### PR TITLE
adding D.Cornet variables and traits

### PR DESCRIPTION
**Pull request by marieALaporte**
Description 
-----------------------------------------------------

Checklist 
---------------------------------------------------------------------------------------
- [ ] An OBO file generated from an OWL file
- [x] New terms
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [ ] Term updates
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [ ] CHECKS
  - [ ] New variables have a parent trait ID 
    - [ ] New variables have methods and scales 
  - [ ] The OBO file has been validated
    - [ ] checked in yambase
    - [ ] checked CropOntology

### CODE REVIEW

File: yam-trait-ontology.obo [hunk 1]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100265
+name: Starch grain size
+namespace: YamTrait
+def: "Mean starch grain size of yam tuber flour" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100266
+name: Microscopy image analysis
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6333]: Needs to be YamMethod namespace

File: yam-trait-ontology.obo [hunk 2]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+[Term]
+id: CO_343:0100263
+name: Tuber flour sugar content
+namespace: YamTrait
+def: "Tuber flour sugar content" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100264
+name: Starch grain size computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6315]: missing definition

File: yam-trait-ontology.obo [hunk 3]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+[Term]
+id: CO_343:0100261
+name: Tuber flour starch content
+namespace: YamTrait
+def: "Tuber flour starch content" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100262
+name: Tuber flour sugar content computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6298]: missing definition

File: yam-trait-ontology.obo [hunk 4]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+namespace: YamTrait
+def: "Prediction of sugar content using NIRS spectra (FOSS 6500) of tuber flour sample" []
+is_a: CO_343:1000021 ! Prediction
+relationship: method_of CO_343:0100258 ! Tuber flour protein content
+relationship: method_of CO_343:0100261 ! Tuber flour starch content
+relationship: method_of CO_343:0100263 ! Tuber flour sugar content
+
+[Term]
+id: CO_343:0100260
+name: Tuber flour starch content computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6281]: missing definition

File: yam-trait-ontology.obo [hunk 5]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100258
+name: Tuber flour protein content
+def: "Tuber flour protein content" []
+namespace: YamTrait
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100259
+name: NIRS model prediction - computation
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6272]: method namespace

File: yam-trait-ontology.obo [hunk 6]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100255
+name: Tuber flesh browning 
+namespace: YamTrait
+def: "Intensity of tuber flesh enzymatic oxidation (degree of flesh color change or browning of cut yam tuber)" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100256
+name: Color difference bewteen mean tuber flesh color computed from standardized image L*a*b* values at cutting and 15 minutes latter 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6247]: YamMethod namespace?

File: yam-trait-ontology.obo [hunk 7]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100253
+name: Mean color difference between average tuber flesh color computed from standardized image L*a*b* values and  tuber flesh color of individual pixel values 
+namespace: YamTrait
+def: "Compute the ∆E2000 between average tuber flesh color computed from standardized image L*a*b* values and  tuber flesh color of individual pixel values " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100241 ! Tuber flesh uniformity
+
+[Term]
+id: CO_343:0100254
+name: Intensity of tuber flesh oxidation computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6229]: add definition

File: yam-trait-ontology.obo [hunk 8]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+is_a: CO_343:1000003 ! Variable
+relationship: variable_of CO_343:0100227 ! uniteless
+relationship: variable_of CO_343:0100241 ! Tuber flesh uniformity
+relationship: variable_of CO_343:0100253 ! Mean color difference between average tuber flesh color computed from standardized image L*a*b* values and  tuber flesh color of individual pixel values
+created_by: D. Cornut
+
+[Term]
+id: CO_343:0100253
+name: Mean color difference between average tuber flesh color computed from standardized image L*a*b* values and  tuber flesh color of individual pixel values 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6222]: YamMethod namespace

File: yam-trait-ontology.obo [hunk 9]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100251
+name: Tuber flesh yellow index computation from standardized image L*a*b* values 
+namespace: YamTrait
+def: "Compute the yellow index based on the flesh surface color of tuber longitudinal section using image analysis with a color correction based on a custom color chart " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100250 ! Yellowness of the Flesh colour
+
+[Term]
+id: CO_343:0100252
+name: Tuber color uniformity computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6211]: add definition

File: yam-trait-ontology.obo [hunk 10]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100250
+name: Yellowness of the Flesh colour
+namespace: YamTrait
+def: "The surface yellowness of the yam tuber flesh" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100251
+name: Tuber flesh yellow index computation from standardized image L*a*b* values 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6204]: YamMethod

File: yam-trait-ontology.obo [hunk 11]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100248
+name: Tuber flesh white index computation from standardized image L*a*b* values 
+namespace: YamTrait
+def: "Compute the white index based on the flesh surface color of tuber longitudinal section using image analysis with a color correction based on a custom color chart " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100247 ! Whitness of the Flesh colour
+
+[Term]
+id: CO_343:0100249
+name: Tuber flesh yellowness computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6186]: add definition

File: yam-trait-ontology.obo [hunk 12]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100247
+name: Whitness of the Flesh colour
+namespace: YamTrait
+def: "The surface whiteness of the yam tuber flesh" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100248
+name: Tuber flesh white index computation from standardized image L*a*b* values 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6179]: namespace: YamMethod

File: yam-trait-ontology.obo [hunk 13]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100245
+name: Tuber flesh brown index computation from standardized image L*a*b* values 
+namespace: YamTrait
+def: "Compute the brown index based on the flesh surface color of tuber longitudinal section using image analysis with a color correction based on a custom color chart " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100244 ! Browness of the Tuber flesh colour
+
+[Term]
+id: CO_343:0100246
+name: Tuber flesh whiteness computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6161]: add definition

File: yam-trait-ontology.obo [hunk 14]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100244
+name: Browness of the Tuber flesh colour
+namespace: YamTrait
+def: "The surface browness of the yam tuber flesh" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100245
+name: Tuber flesh brown index computation from standardized image L*a*b* values 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6154]: YamMethod

File: yam-trait-ontology.obo [hunk 15]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100242
+name: Color difference bewteen mean tuber head and tail flesh color computed from standardized image L*a*b* values 
+namespace: YamTrait
+def: "Compute the ∆E2000 bewteen mean tuber head and tail flesh color computed from standardized image L*a*b* values " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100241 ! Tuber flesh uniformity
+
+[Term]
+id: CO_343:0100243
+name: Tuber flesh brownness computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6136]: add definition

File: yam-trait-ontology.obo [hunk 16]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100241
+name: Tuber flesh uniformity 
+namespace: YamTrait
+def: "Uniformity of tuber flesh surface color in longitudinal section" []
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100242
+name: Color difference bewteen mean tuber head and tail flesh color computed from standardized image L*a*b* values 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6129]: YamMethod

File: yam-trait-ontology.obo [hunk 17]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100239
+name: Tuber flesh color computation from standardized image L*a*b* values 
+namespace: YamTrait
+def: "Compute the average of the flesh surface color of tuber longitudinal section using image analysis with a color correction based on a custom color chart " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100238 ! Tuber flesh color
+
+[Term]
+id: CO_343:0100240
+name: Tuber flesh color gradient computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6111]: add definition

File: yam-trait-ontology.obo [hunk 18]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100238
+name: Tuber flesh color 
+def: "The surface color of the yam tuber flesh" []
+namespace: YamTrait
+is_a: CO_343:1000007 ! Quality_trait
+
+[Term]
+id: CO_343:0100239
+name: Tuber flesh color computation from standardized image L*a*b* values 
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:6104]: namespace: YamMethod

File: yam-trait-ontology.obo [hunk 19]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100236
+name: Duration between 5% emergence and 95% senescence of subplot - Computation
+namespace: YamTrait
+def: "Averaged growth cycle duration expressed in days between  5% emergence and 95% senescence (i.e. 5% of maximum ground cover after time to maximum ground cover) in two subplots.\nGround cover computed from weekly images analysis. \nMedian senescence computed from inflection point value using log-logistic time-to-event modeling of weekly sprout counts expressed in days after planting.\nTime at 5% of maximum growth after time at maximum growth computed via inverse regression of time to event Beta ground cover dynamic model." []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100235 ! Growth cycle length
+
+[Term]
+id: CO_343:0100237
+name: Tuber flesh color computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:6086]: add definition

File: yam-trait-ontology.obo [hunk 20]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100233
+name: Duration between 5 and 95% senescence of subplot - Computation
+namespace: YamTrait
+def: "Averaged senescence period duration expressed in days between 5 and 95% senescence (i.e. 95%  and 5% of maximum ground cover after time to maximum ground cover) in two subplots.\nGround cover computed from weekly images analysis. \nTime at 95% and 5% of maximum growth after time at maximum growth computed via inverse regression of time to event Beta ground cover dynamic model." []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100232 ! Senescence duration
+
+[Term]
+id: CO_343:0100234
+name: Growth cycle length computation in days between 5% emergence and 95% senescence
```
Comment (nmenda) [inline yam-trait-ontology.obo:6061]: add definition

File: yam-trait-ontology.obo [hunk 21]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100230
+name: Duration from median emergence to 5% senescence of subplot - Computation
+namespace: YamTrait
+def: "Averaged senescence earliness expressed in days from median emergence to 5% senescence (i.e. 95% of maximum ground cover after time to maximum ground cover) in two subplots.\nGround cover computed from weekly images analysis. \nTime at 95% of maximum growth after time at maximum growth computed via inverse regression of time to event Beta ground cover dynamic model." []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100229 ! Senescence earliness
+
+[Term]
+id: CO_343:0100231
+name: Senescence period duration computation in days between 5 and 95% senescence
```
Comment (nmenda) [inline yam-trait-ontology.obo:6036]: add definition

File: yam-trait-ontology.obo [hunk 22]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100223
+name: Duration from median emergence to 80% of maximum ground cover of subplot - Computation
+namespace: YamTrait
+def: "Averaged crop vigor expressed in days from median emergence to 80% of maximum ground cover in subplots.\nGround cover computed from weekly images analysis. \nTime at 80% of maximum growth computed via inverse regression of time to event Beta ground cover dynamic model. " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100222 ! Crop vigor
+
+[Term]
+id: CO_343:0100224
+name: Crop cover compactness computation
```
Comment (nmenda) [inline yam-trait-ontology.obo:5972]: add definition

File: yam-trait-ontology.obo [hunk 23]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100220
+name: Subplot maximum ground cover from weekly image analysis - Computation
+namespace: YamTrait
+def: "Averaged maximum ground cover over two subplots. \nGround cover computed from weekly images analysis. \nMaximum ground cover computed from ground cover dynamic modeled using time to event Beta Growth model. " []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100219 ! Maximum ground cover
+
+[Term]
+id: CO_343:0100221
+name: Crop vigor computation in days from median emergence to 80% of maximum ground cover
```
Comment (nmenda) [inline yam-trait-ontology.obo:5947]: add definition

File: yam-trait-ontology.obo [hunk 24]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100219
+name: Maximum ground cover
+namespace: YamTrait
+def: "Maximum ground cover" []
+is_a: CO_343:1000008 ! Agronomical_trait
+
+[Term]
+id: CO_343:0100220
+name: Subplot maximum ground cover from weekly image analysis - Computation
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:5940]: namespace: YamMethod

File: yam-trait-ontology.obo [hunk 25]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+name: Nodes carrying cataphylles  - Counting
+namespace: YamTrait
+def: "Compute the average of the number of nodes carrying cataphylles on the main stem of 5 plants choosen randomly" []
+is_a: CO_343:1000012 ! Counting
+relationship: method_of CO_343:0100212 ! Number of nodes carrying cataphylles
+
+[Term]
+id: CO_343:0100214
+name: Nodes carrying cataphylle per plant
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:5891]: namespace: YamScale

File: yam-trait-ontology.obo [hunk 26]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100212
+name: Number of nodes carrying cataphylles
+namespace: YamTrait
+def: "Mean number of nodes carrying cataphylles on main stem of 5 plants choosen randomly" []
+is_a: CO_343:1000010 ! Morphological_trait
+
+[Term]
+id: CO_343:0100213
+name: Nodes carrying cataphylles  - Counting
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:5883]: namespace: YamMethod

File: yam-trait-ontology.obo [hunk 27]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100210
+name: Presence of cataphylle
+namespace: YamTrait
+def: "Observe the presence or absence of cataphylle" []
+is_a: CO_343:1000014 ! Estimation
+relationship: method_of CO_343:0100209 ! Heterophylly presence
+
+[Term]
+id: CO_343:0100211
+name: Number of nodes carrying cataphylles before first true leaf counting/plant
```
Comment (nmenda) [inline yam-trait-ontology.obo:5865]: add definition

File: yam-trait-ontology.obo [hunk 28]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100207
+name: Period from 5% to 95% emergence
+namespace: YamTrait
+def: "Computation of the substraction between the necessary time before 95% and 5% emergence using after-fitting inverse regression of log-logistic time-to-event modeling of weekly sprout counts expressed in days after planting" []
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100206 ! Emergence vigor
+
+[Term]
+id: CO_343:0100208
+name: Heterophylly presence estimation nominal 0&1
```
Comment (nmenda) [inline yam-trait-ontology.obo:5840]: add definition

File: yam-trait-ontology.obo [hunk 29]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100206
+name: Emergence vigor
+namespace: YamTrait
+def: "Elapsed time between time of 5th and 95th percentiles emergence sprouts in a plot" []
+is_a: CO_343:1000008 ! Agronomical_trait
+
+[Term]
+id: CO_343:0100207
+name: Period from 5% to 95% emergence
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:5833]: namespace: yamMethod

File: yam-trait-ontology.obo [hunk 30]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100204
+name: Period from planting to 5% emergence
+namespace: YamTrait
+is_a: CO_343:1000013 ! Computation
+relationship: method_of CO_343:0100203 ! Latency before emergence
+def: "Computation of the necessary time before 5% emergence using after-fitting inverse regression of log-logistic time-to-event modeling of weekly sprout counts expressed in days after planting" []
+
+[Term]
+id: CO_343:0100205
+name: Emergence vigor computation in number of days between 5% and 95% sprouts emergence
```
Comment (nmenda) [inline yam-trait-ontology.obo:5815]: add definition

File: yam-trait-ontology.obo [hunk 31]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100203
+name: Latency before emergence
+namespace: YamTrait
+is_a: CO_343:1000006 ! Phenological_trait
+def: "Elapsed time between planting date and date of 5th percentile emergence sprouts in a plot" []
+
+[Term]
+id: CO_343:0100204
+name: Period from planting to 5% emergence
+namespace: YamTrait
```
Comment (nmenda) [inline yam-trait-ontology.obo:5808]: namespace: YamMethod

File: yam-trait-ontology.obo [hunk 32]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
...
+id: CO_343:0100201
+name: Period from planting to median emergence
+namespace: YamTrait
+is_a: CO_343:1000013 ! Computation
+def: "Computation of inflection point value using log-logistic time-to-event modeling of weekly sprout counts expressed in days after planting" []
+relationship: method_of CO_343:0100200 ! Earliness of emergence
+
+[Term]
+id: CO_343:0100202
+name: Latency period duration computation in number of days from planting to 5% sprouts emergence
```
Comment (nmenda) [inline yam-trait-ontology.obo:5790]: add definition

File: yam-trait-ontology.obo [hunk 33]
```diff
@@ -5760,6 +5760,588 @@ namespace: YamScale
 is_a: CO_343:1000019 ! Numerical
 relationship: scale_of CO_343:0010169 ! number of leaves - computation
 
+[Term]
+id: CO_343:0100199
+name: Earliness of emergence computation in number of days from planting to 50% sprouts emergence
```
Comment (nmenda) [inline yam-trait-ontology.obo:5765]: add definition

**Review by nmenda — CHANGES_REQUESTED** [review]

**Comment by nmenda** [comment]
@marieALaporte the variables need a definition, and the new methods need to have a YamMethod namespace. I'll look for the new commit and then test it in YamBase.

File: yam-trait-ontology.obo (hunk 33 above)
Comment (marieALaporte) [reply yam-trait-ontology.obo:5765]: done

**Review by marieALaporte — COMMENTED** [review]
I fixed the namespaces for methods and scales and added definitons for variables

**Review by nmenda — APPROVED** [review]
Looks good in YamBase!

**Pull request merged**